### DISCUSSION TITLE
xmlsec1-1.2.37 to xmlsec1-1.3.0 migration

### DISCUSSION THREAD

**Discussion by tfloodeenjr** [lsh123/xmlsec#671] [Q&A] [answered]
We are attempting to migrate from xmlsec 1.2.37 to 1.3.0. We use OpenSSL 3.1.0 and libxml2 2.10.3. Everything seems to work fine with xmlsec 1.2.37, but when we go to 1.3.0, the build works, but runtime fails with these debug messages. Any ideas?

func=xmlSecBase64CtxUpdate_ex:file=..\src\base64.c:line=255:obj=unknown:subj=xmlSecBase64CtxEncode:error=1:xmlsec library function failed:
func=xmlSecBase64Execute:file=..\src\base64.c:line=898:obj=base64:subj=xmlSecBase64CtxUpdate_ex:error=1:xmlsec library function failed:
func=xmlSecTransformDefaultPushBin:file=..\src\transforms.c:line=1971:obj=base64:subj=xmlSecTransformExecute:error=1:xmlsec library function failed:final=0
func=xmlSecTransformDefaultPushBin:file=..\src\transforms.c:line=1995:obj=base64:subj=xmlSecTransformPushBin:error=1:xmlsec library function failed:final=0;outSize=16016
func=xmlSecTransformIOBufferWrite:file=..\src\transforms.c:line=2580:obj=aes128-cbc:subj=xmlSecTransformPushBin:error=1:xmlsec library function failed:
I/O error : write error
func=xmlSecBase64CtxUpdate_ex:file=..\src\base64.c:line=255:obj=unknown:subj=xmlSecBase64CtxEncode:error=1:xmlsec library function failed:
func=xmlSecBase64Execute:file=..\src\base64.c:line=898:obj=base64:subj=xmlSecBase64CtxUpdate_ex:error=1:xmlsec library function failed:
func=xmlSecTransformDefaultPushBin:file=..\src\transforms.c:line=1971:obj=base64:subj=xmlSecTransformExecute:error=1:xmlsec library function failed:final=1
func=xmlSecTransformDefaultPushBin:file=..\src\transforms.c:line=1995:obj=base64:subj=xmlSecTransformPushBin:error=1:xmlsec library function failed:final=1;outSize=16032
func=xmlSecTransformIOBufferClose:file=..\src\transforms.c:line=2600:obj=aes128-cbc:subj=xmlSecTransformPushBin:error=1:xmlsec library function failed:
func=xmlSecEncCtxXmlEncrypt:file=..\src\xmlenc.c:line=402:obj=unknown:subj=xmlOutputBufferClose:error=5:libxml2 library function failed:xml error: 1546: write error
Error: encryption failed
Error : Can't write <filename redacted> file!

Here is the configuration information:

libxmlsec version: 1.3.0
Created Makefile.

XMLSEC configuration
----------------------------
         Use Crypto: openssl
 Use Default Crypto: openssl
        Use OpenSSL: Enabled
Use OpenSSL Version: 300
            Use NSS: Disabled
       Use MSCrypto: Disabled
          Use MSCng: Disabled
        Use LibXSLT: Disabled
          Use iconv: Disabled
         Use size_t: Disabled
  Use legacy crypto: Enabled

Win32 build configuration
-------------------------
     NT 4.0 support: Disabled
  C-Runtime option: /MDd
           Unicode: Disabled
     Debug symbols: Enabled
          Memcheck: Disabled
Warnings as errors: Disabled
          Pedantic: Enabled
        C compiler: cl.exe
  C compiler flags: 
Static xmlsec libs: Enabled
 Enable DL support: Disabled
    Install prefix: C:\install
      Put tools in: $(PREFIX)\bin
    Put headers in: $(PREFIX)\include
Put static libs in: $(PREFIX)\lib
Put shared libs in: $(PREFIX)\bin
      Include path: C:\install\include;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\ATLMFC\include;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\include;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt
          Lib path: C:\install\lib;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\ATLMFC\lib\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\lib\x64;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64;C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\ucrt\x64;C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64

Crypto configuration
-------------------------
Use OpenSSL3 Engine: Disabled

Thanks,
-Tom

**Comment by lsh123** [comment, marked as answer]
Could you please try 1.3.1? This might be same problem fixed in https://github.com/lsh123/xmlsec/pull/656

**Reply by tfloodeenjr** [reply]
It looks like that works so far. thanks!

**Reply by lsh123** [reply]
Thanks for letting me know!
